pkg/commands/helpers/kubectl_auth.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog"
+	"k8s.io/kops/cmd/kops/util"
+	"k8s.io/kops/pkg/commands/commandutils"
+	"k8s.io/kops/pkg/pki"
+	"k8s.io/kops/pkg/rbac"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kubectl/pkg/util/i18n"
+)
+
+var (
+	kubectlAuthShort = i18n.T(`Helper for kubectl authentication integration`)
+)
+
+// HelperKubectlAuthOptions holds the options for generating an authentication token
+type HelperKubectlAuthOptions struct {
+	// ClusterName is the name of the cluster we are targeting
+	ClusterName string
+
+	// Duration specifies the desired duration of the credential
+	Duration time.Duration
+
+	// APIVersion specifies the version of the client.authentication.k8s.io schema in use
+	APIVersion string
+}
+
+// InitDefaults populates the default values of options
+func (o *HelperKubectlAuthOptions) InitDefaults() {
+	o.Duration = 1 * time.Hour
+}
+
+// NewCmdHelperKubectlAuth builds a cobra command for the kubectl-auth command
+func NewCmdHelperKubectlAuth(f *util.Factory, out io.Writer) *cobra.Command {
+	options := &HelperKubectlAuthOptions{}
+	options.InitDefaults()
+
+	cmd := &cobra.Command{
+		Use:   "kubectl-auth",
+		Short: kubectlAuthShort,
+		Run: func(cmd *cobra.Command, args []string) {
+			ctx := context.TODO()
+
+			err := RunKubectlAuthHelper(ctx, f, out, options)
+			if err != nil {
+				commandutils.ExitWithError(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&options.APIVersion, "api-version", options.APIVersion, "version of client.authentication.k8s.io schema in use")
+	cmd.Flags().StringVar(&options.ClusterName, "cluster", options.ClusterName, "cluster to target")
+	cmd.Flags().DurationVar(&options.Duration, "validity", options.Duration, "attempt to issue a credential with this validity")
```

Comment:
```suggestion
	cmd.Flags().DurationVar(&options.Duration, "validity", options.Duration, "lifetime of the credential to issue")
```

Reply:
I'd try to come up with a better flag name than `--validity`. validity on its own doesn't suggest a duration, merely that something is valid. Maybe just `--duration` ?

Reply:
`--lifetime` would be my suggestion

Reply:
Will use the help text and switch to `lifetime`

BTW these flags are not very visible; the whole helpers command subtree is hidden.  But it is visible in your kubeconfig if you're using the plugin, so it's worth having them be clear.